solutions/src/main/java/ru/sberbank/school/task06/CollectionUtils.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    /**
+     * Проверка на содержение одной коллекции в другой.
+     *
+     * @param c1 первая коллекция
+     * @param c2 вторая коллекция
+     * @return true, если все элементы коллекции c2 содержатся в c1
+     */
+    public static <T> boolean containsAll(List<? super T> c1, List<? extends T> c2) {
+        return c1.containsAll(c2);
+    }
+
+    /**
+     * Проверка на содержение хотя бы одного элемента одной коллекции в другой.
+     *
+     * @param c1 первая коллекция
+     * @param c2 вторая коллекция
+     * @return true, если хотя бы 1 элемент коллекции c2 содержатся в c1
+     */
+    public static <T> boolean containsAny(List<? super T> c1, List<? extends T> c2) {
+        for (Object o : c1) {
+            for (T t : c2) {
+                if (o.equals(t)) {
+                    return true;
+                }
+
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Создание коллекции, содержащей элементы из входной коллекции в диапазоне от min до max (не
+     * индексы, а значения). Элементы необходимо сравнивать, через интерфейс Comparable Пример:
+     * range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6) вернет {3,4,5,6}
+     *
+     * @param list исходная коллекция
+     * @param min  минимальное значение
+     * @param max  максимальнео значение
+     * @return отфильрованный по минимальному и максимальному значению список
+     */
+    public static <T extends Comparable<? super T>> List<T> range(List<? extends T> list, T min, T max) {
+//        TreeSet<T> set = new TreeSet<T>() {{
+//                for (T t : list) {
+//                    if (t.compareTo(min) >= 0 && t.compareTo(max) <= 0) {
+//                        add(t);
+//                    }
+//                }
+//            }};
+
+        return list.stream()
+                .filter((T t) -> t.compareTo(min) >= 0 && t.compareTo(max) <= 0)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Создание коллекции, содержащей элементы из входной коллекции в диапазоне от min до max (не
+     * индексы, а значения). Элементы необходимо сравнивать, через интерфейс Comparator Пример:
+     * range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6) вернет {3,4,5,6}
+     *
+     * @param list исходная коллекция
+     * @param min  минимальное значение
+     * @param max  максимальнео значение
+     * @return отфильрованный по минимальному и максимальному значению список
+     */
+    public static <T> List<T> range(List<? extends T> list, T min, T max, Comparator<T> comparator) {
+//        TreeSet<T> set = new TreeSet<T>() {{
+//                for (T t : list) {
+//                    if (comparator.compare(t, min) >= 0 && comparator.compare(t, max) <= 0) {
+//                        add(t);
+//                    }
+//                }
+//
+//            }};
+
+        return list.stream()
+                .filter((T t) ->comparator.compare(t, min) >= 0 && comparator.compare(t, max) <= 0)
```

Comment:
WhitespaceAround: &#39;-&gt;&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a277877688>*